CleverRaven/Cataclysm-DDA
# Removed force_unarmed from Taekwondo

#### Summary
Balance "Removed force_unarmed from Taekwondo"


#### Purpose of change


Removes force_unarmed from Taekwondo and adds prevent_weapon_blocking attrbute

#### Describe the solution



With the merging of #53954, it is no longer necessary for Taekwondo to have a unique attribute to force unarmed techniques. These techniques can occur naturally now if they are given set their `allow_melee` attribute to true. However, the style isn't allow to block with held items so a new attribute `prevent_weapon_blocking` has been added. All styles except Taewondo and "Force unarmed" have this set as false which means you can block with a weapon if the style allows that weapon.

#### Describe alternatives you've considered


I considered removing `force_unarmed` completely but I found that the "Force unarmed" style also uses the attribute. While niche, I didn't want to prevent the player from being able to attack unarmed while holding a weapon if they so choose. But unlike Taekwondo, "Force unarmed" doesn't have any techniques which makes `force_unarmed` necessary to prevent attacking with a weapon. So I just added `prevent_weapon_blocking` to the "Force unarmed" style to keep weapon blocking from occurring instead.


#### Testing



- Set your martial art style to Taekwondo and equip any weapon.
- Allow any enemy to attack you. You will never block an attack with your weapon.
- Attack while holding your weapon. You will see all Taekwondo techniques work as normal.
- Change to a different style that has your equipped weapon as style weapon.
- Allow any enemy to attack you. You should be able to block attacks with your weapon.
